reclass/datatypes/parameters.py

```diff
@@ -119,7 +120,25 @@ def _extend_list(self, cur, new, path):
             ret.append(self._merge_recurse(None, new[i], path.new_subpath(offset + i)))
         return ret
 
-    def _merge_dict(self, cur, new, path):
+    def _merge_dict(self, cur, new, path, initmerge):
+        """Merge a dictionary with another dictionary.
+
+        Iterate over keys in new. If this is not an initialization merge and
+        the key begins with PARAMETER_DICT_KEY_OVERRIDE_PREFIX, override the
+        value of the key in cur. Otherwise deeply merge the contents of the key
+        in cur with the contents of the key in _merge_recurse over the item.
+
+        Args:
+            cur (dict): Current dictionary
+            new (dict): Dictionary to be merged
+            path (string): Merging path from recursion
+            initmerge (bool): True if called as part of entity init
```

Comment:
The merge is actually performed by Parameters.__init__ which happens to be called by Entity.__init__. Lines 47-50 of Parameters.__init__...
```
        if mapping is not None:
            # we initialise by merging, otherwise the list of references might
            # not be updated
            self.merge(mapping)
```
In actuality this happens, as far as I can tell, every time.